### ISSUE TITLE
Android Auto Wireless support

### ISSUE THREAD

**Issue by MrNessuno** [sn-00-x/aa4mg#13]
I successfully connected the phone via USB cable but my car also support Android Auto wireless. I tried connecting with that, the car is listed in the Android Auto app,  but when I try to connect from the car the android auto app keep restating without successfully connecting.
Am I doing something wrong?

**Comment by panoskpv** [comment]
I tried as per the instructions and disabled all network data from aa and wireless not working. When I allowed mobile and background data wireless is working. Is the denial of data usefull? is the app connecting to google even if i use XPrivacyLua and restrict all?

**Comment by knuxyl** [comment]
When I try wireless connection, it says to continue on the Android Auto device but nothing shows up on the screen.

**Comment by knuxyl** [comment]
Sorry i forgot to post back, but apparently whenever a VPN is in use Android Auto won't work. That's what was causing my issue. It has something to do with what Android is using for the internet. I have no idea what to do about it, it's apparently a known bug.

**Comment by bioluks** [comment]
Thanks for getting back so quickly, yes I have an app called TrackerControl in the background, which is like many a fake VPN that does domain/DNS blocking on a per-app basis. Never thought about disabling it alltogether.

So with a disabled VPN, on a custom ROM with microG, and allowed background traffic for Android Auto everything just works fine for you @knuxyl ?

**Comment by knuxyl** [comment]
> Thanks for getting back so quickly, yes I have an app called TrackerControl in the background, which is like many a fake VPN that does domain/DNS blocking on a per-app basis. Never thought about disabling it alltogether.
> 
> So with a disabled VPN, on a custom ROM with microG, and allowed background traffic for Android Auto everything just works fine for you @knuxyl ?

yeah everything worked as expected and i didnt use the lsposed module because i dont have any apps that didnt show up that i wanted. youtube didnt show up with or without the module so i didnt use it.

im using lineageos 21 for microg on a pixel 5 and typically use a wireguard vpn. with vpn turned off everything worked as expected.

**Comment by panoskpv** [comment]
And how about any data end up to Google?

**Comment by bioluks** [comment]
> And how about any data end up to Google?

I have explained the options one can go for in Pull-Request #24 . You can read it [here](https://github.com/bioluks/aa4mg/tree/wireless-documentation#installation).

Basically if you allow traffic (which you have to for Android Auto wireless to work) it WILL contact external servers. Question is how much you want Android Auto wireless. You could experiment a bit with root firewall solutions like AFWall and disallow traffic from there while still allowing in Android settings. That _could_ work, didn't test it myself though.

**Comment by prgreadonly** [comment]
Is wireless android auto working with this module and recent versions of Android Auto?  I haven't been able to get it to work with my Sony VAX-AX6000:
When I enable WIreless Android Auto on my OnePlus 11 running LineageOS/microG and USB is disconnected, I get the notification on my phone to check my stereo to connect to android auto, but my stereo doesn't see my phone. I only ever see the icon for Android Auto when I have a USB cable connected.
Could anyone let me know if WIreless Android Auto is working for them with aa4mg and if so, what's your setup?

**Comment by bioluks** [comment]
@prgreadonly If you went through the steps in the README skip step 5, since Android Auto Wireless uses Wi-Fi in the background to connect to the Headunit of your car. With step 5 applied you can only use USB connection.

Like mentioned in this thread as well, a VPN somehow messes with the networking configuration Android Auto needs. You can turn off your VPN or fully exclude Android Auto if the app you use has such an option.

That's why I published a pull-request trying to be more clear about this in the documentation but I guess this project will have no more activity for the future.

In some rare cases (some specific Volkswagen cars) the headunit was produced with a bug that prevents them from connecting wirelessly unless you force restart the unit. You can do that by holding down the power button until it reboots (10-15 seconds generally).

**Comment by prgreadonly** [comment]
Thanks for the additional info @bioluks !  I had already seen your documentation PR and followed those instructions, but I'm still not able to connect with Android Auto.
I'm going to do some more troubleshooting with my Oneplus 11 to try to determine if the issue i'm encountering is specific to aa4mg or if I also have this issue when I have google play services installed instead of microG/aa4mg

**Comment by Klemet** [comment]
Hey there !

Sorry for digging up the issue; I've been obliged to consider wireless android auto since my USB C port has some trouble, but I haven't been able to make it work.

- Tried with another phone (with a regular android installation) : It works flawlessly (so my car head unit looks to be OK)
- Got the car to very temporarily detected android auto from USB (but my USB port having trouble, this did not work further than this); so it seems like it's properly installed on my phone.
- Tried on my phone with Android Auto installed according to instruction + AFWall completly disabled or app allowed + all network permissions given to the app + VPN disabled. Alas, the car head unit doesn't propose the Android auto button, even though android auto on my phone does detect a wireless Android Auto connection as possible. See screenshot :

<img width="386" height="831" alt="Image" src="https://github.com/user-attachments/assets/d127eedc-c49e-44c3-8863-67068361246e" />

If anybody has suggestions on how to fix this, I'm all ears !
